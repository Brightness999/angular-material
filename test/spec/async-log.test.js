/**
 * @fileoverview Test asynchronous logging.
 */
const Logality = require('../..');
const { sink, stubLogality } = require('../lib/tester.lib');

describe('Asynchronous Logging', () => {
  stubLogality();

  test('Async logging with default writestream to stdout', () => {
    const logality = new Logality({
      appName: 'testLogality',
      async: true,
    });

    const log = logality.get();

    log('info', 'hello world');
  });

  test('Async Logging with async writabble stream', async () => {
    const logality = new Logality({
      appName: 'testLogality',
      wstream: sink((chunk, enc, cb) => {
        expect(chunk).toMatchSnapshot();
        expect(enc).toEqual('utf8');
        cb();
      }),
    });

    const log = logality.get();

    await log('info', 'hello world', { custom: { a: 1, b: 2 } });
  });

  test('Async Logging Streamer Error Propagates', async () => {
    const logality = new Logality({
      appName: 'testLogality',
      wstream: sink((chunk, enc, cb) => {
        expect(chunk).toMatchSnapshot();
        expect(enc).toEqual('utf8');
        const err = new Error('420');
        cb(err);
      }),
    });

    const log = logality.get();

    let errorThrown = false;

    try {
      await log('info', 'hello world', { custom: { a: 1, b: 2 } });
    } catch (ex) {
      expect(ex.context.message).toEqual('420');
      errorThrown = true;
    }

    expect(errorThrown).toEqual(true);
  });
});
