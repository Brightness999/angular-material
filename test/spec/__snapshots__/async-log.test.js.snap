// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asynchronous Logging Async Logging Streamer Error Propagates 1`] = `
Object {
  "context": Object {
    "custom": Object {
      "a": 1,
      "b": 2,
    },
    "runtime": Object {
      "application": "testLogality",
    },
    "source": Object {
      "file_name": "/test/spec/async-log.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello world",
  "severity": 6,
}
`;

exports[`Asynchronous Logging Async Logging with async writabble stream 1`] = `
Object {
  "context": Object {
    "custom": Object {
      "a": 1,
      "b": 2,
    },
    "runtime": Object {
      "application": "testLogality",
    },
    "source": Object {
      "file_name": "/test/spec/async-log.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello world",
  "severity": 6,
}
`;
